### PR TITLE
Feature/setup validation errors

### PR DESCRIPTION
**Pull request by bimal125**
## Changes

* Add good practice mutation

Mention related users here if any.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/good_practice/models.py [hunk 1]
```diff
@@ -112,10 +112,18 @@ class StageType(models.TextChoices):
     start_year = models.BigIntegerField(verbose_name=_('Start year'))
     end_year = models.BigIntegerField(blank=True, null=True, verbose_name=_('End year'))
     page_viewed_count = models.BigIntegerField(default=0, verbose_name=_('Total page viewed count'))
+    is_public = models.BooleanField(
+        editable=False, default=False, verbose_name=_('Is public?'),
+        help_text=_('When a public submits a good practice it should be true which is immutable.'),
```
Comment (thenav56) [inline apps/good_practice/models.py:117]:
```suggestion
        help_text=_('It indicates if a good practice is submitted without authentication'),
```

File: apps/good_practice/tests/factories.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import datetime
+from factory import fuzzy
+from factory.django import DjangoModelFactory
+
+from apps.good_practice.models import GoodPractice
+
+
+class GoodFactoryFactory(DjangoModelFactory):
```
Comment (bewakes) [inline apps/good_practice/tests/factories.py:8]: Perhaps the name should be `GoodPracticeFactory`? I am not entirely sure though.

**Review by thenav56 — CHANGES_REQUESTED** [review]
Few comments.
Also, the test seems to be WIP.

**Review by bewakes — APPROVED** [review]

**Review by k9845 — APPROVED** [review]

File: apps/good_practice/tests/factories.py (hunk 2 above)
Comment (k9845) [reply apps/good_practice/tests/factories.py:8]: @bimal125 look into this.

**Review by k9845 — APPROVED** [review]

**Review by thenav56 — APPROVED** [review]
LGTM.

**Review by thenav56 — APPROVED** [review]

**Pull request merged**